tests/test_post_to_studio.py

```diff
@@ -464,3 +464,17 @@ def test_post_to_studio_skip_if_no_repo_url(
         live.next_step()
 
     assert mocked_post.call_count == 0
+
+
+def test_post_to_studio_increment_internal_state(
```

Comment:
we should not be testing an internal state, rather the actual bug / scenario that you we are trying to solve / prevent

tbh, I don't quite understand it from the description / I don't have enough context.

could you describe it and / or write an actual test that would cover it? it will be easier to review it then + it'll force us to have a proper test for this